File: frontend/src/interfaces/channel.ts
interface Channel {
    id: number,
    code: string,
    name: string,
    curPlayers: number,
    maxPlayers: number
}
interface ChannelChat {
    nickname: string,
    uuid: string,
    content: string,
}

interface RecieveChannelChat {
    nickname: string,
    uuid: string,
    content: string,
    createdDate: string,    
}

File: backend/src/main/java/com/ssafy/apm/channel/domain/Channel.java
package com.ssafy.apm.channel.domain;

import lombok.*;
import jakarta.persistence.*;

@Entity
@Getter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "channel")
public class Channel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String code;
    private String name;
    private Integer curPlayers;
    private Integer maxPlayers;

    public boolean checkChannelAccess() {
        return curPlayers < maxPlayers;
    }

    public void decreaseCurPlayers(Integer count) {
        this.curPlayers = count;
    }

    public void increaseCurPlayers(Integer count) {
        this.curPlayers = count;
    }


}
